src/modules/auth/BaseSig.sol

```diff
@@ -39,6 +39,14 @@ library BaseSig {
   error UnusedSnapshot(Snapshot _snapshot);
   /// @notice Error thrown when the signature flag is invalid
   error InvalidSignatureFlag(uint256 _flag);
+  /// @notice Error thrown when a chained signature is nested inside another chained signature
+  error ChainedSignatureNestedInChainedSignature();
+
+  enum RecoverMode {
+    Initial,
+    UseProvidedCheckpointer,
+    IgnoreReadCheckpointer
```

Comment:
Can we rename this to `SkipSnapshotRead`? Read (reed) and read (red) is ambiguous. We don't ignore the reed checkpointer flow, we do ignore the red checkpointer.